compiler-cli/src/lsp.rs

```diff
@@ -167,14 +167,38 @@ impl LanguageServer {
                 }
 
                 lsp_server::Message::Notification(notification) => {
-                    if let Err(error) = self.handle_notification(&connection, notification) {
-                        match error_to_diagnostic(&error) {
+                    match self.handle_notification(&connection, notification) {
+                        Ok(()) => {
+                            let mut diagnostics: HashMap<PathBuf, Vec<Diagnostic>> = HashMap::new();
```

Comment:
We still want to get warnings even if compilation failed, so let's move this out of the `if let/match` so that it always runs.

Reply:
My initial approach was like this, but the compiler does not return warnings if some error occurs. I don't if it's related to the order they occur, I mean, the compiler returns only the warnings before the error.

Reply:
Done!